src/local/base_node.rs

```diff
@@ -453,7 +453,242 @@ mod tests {
         }
     }
 
-    // TODO: test that returns an error when the lookup table returns an error during search at any level.
-    // TODO: test that when the exact target is found, it returns the correct level and identifier.
-    // TODO: concurrent tests for search_by_id to ensure thread safety and correctness under concurrent access.
+    /// Tests the `search_by_id` method of a `LocalNode` under concurrent conditions where multiple
+    /// threads perform searches in the left direction (`Direction::Left`) simultaneously.
+    ///
+    /// The test:
+    /// - Creates a `LocalNode` with a random identifier, random membership vector,
+    ///   and a lookup table (`RandomLookupTable`).
+    /// - Randomly generates a target identifier to search for.
+    /// - Spawns 20 threads that conduct searches concurrently from the node.
+    ///
+    /// ### Test Specific Logic:
+    /// - Each thread constructs a search request targeting the same identifier and executes the `search_by_id` method.
+    /// - The expected search result is derived by finding the closest matching identifier from the
+    ///   left neighbors in the lookup table (`lt`) that meets the search criteria (e.g., level,
+    ///   target identifier comparison).
+    /// - If no valid neighbor is found, it expects the result to default to the `LocalNode`'s own identifier.
+    ///
+    /// ### Assertions:
+    /// - If a valid neighbor exists, the search output should match both the level and identifier.
+    /// - If no valid neighbor exists, the search result should match the node's own identifier at level 0.
+    #[test]
+    fn test_search_by_id_concurrent_found_left_direction() {
+        let lt = random_lookup_table_with_extremes(LOOKUP_TABLE_LEVELS);
+        let target = random_identifier();
+
+        let node = Arc::new(LocalNode {
+            id: random_identifier(),
```

Comment:
To ensure that the test scenario does not trigger the quality case, that is, the target is expected to be among the left neighbors of the node, if any, but not the node itself,  it is prudent to check that the node’s `id` is not equal to the target's.

Reply:
https://github.com/thep2p/skipgraph-rust/pull/21/commits/ec2236880986d93a1a243a0f26c0527bf9d5b1a8